### PR TITLE
Az scan set kv access policy

### PR DESCRIPTION
**Pull request by VaclavK**

### CODE REVIEW

**Comment by djfr** [comment]
+ @cjfarrelly @rootik

File: src/DevOpsFlex.Automation.PowerShell/AzureKeyvaultHelpers.ps1 [hunk 1]
```diff
@@ -386,4 +386,65 @@ Copies Secrets and Access Policies from the KeyVault my-source-kv in Resource Gr
...
+        [parameter(Mandatory=$true, Position=0)]
+        [string] $IdentityObjectId
+    )
+    $envs =  @{SubName="ci"; Token="ci"; Regions="we"}, 
+    @{SubName="test"; Token="test"; Regions="we", "eus"},
+    @{SubName="sandbox"; Token="sand"; Regions="we", "eus"},
+    @{SubName="preprod"; Token="prep"; Regions="we", "eus"},
+    @{SubName="prod"; Token="prod"; Regions="we", "eus"}
+
+	$domains = "tooling", "payments", "logistics", "checkout", "nikesnkrs", "tahoe"
```
Comment (cjfarrelly) [inline src/DevOpsFlex.Automation.PowerShell/AzureKeyvaultHelpers.ps1]: security domain?

File: src/DevOpsFlex.Automation.PowerShell/AzureKeyvaultHelpers.ps1 [hunk 2]
```diff
@@ -386,4 +386,65 @@ Copies Secrets and Access Policies from the KeyVault my-source-kv in Resource Gr
...
+ensures re 
+#>
+
+    [CmdletBinding()]
+    param
+    (
+        [parameter(Mandatory=$true, Position=0)]
+        [string] $IdentityObjectId
+    )
+    $envs =  @{SubName="ci"; Token="ci"; Regions="we"},
```
[the remark this answers is not in the document]
Comment (VaclavK) [reply src/DevOpsFlex.Automation.PowerShell/AzureKeyvaultHelpers.ps1]: that was the idea of controlled process here...

I am certainly in favour of defining this once... are you suggesting dependency module??

File: src/DevOpsFlex.Automation.PowerShell/AzureKeyvaultHelpers.ps1 (hunk 1 above)
Comment (VaclavK) [reply src/DevOpsFlex.Automation.PowerShell/AzureKeyvaultHelpers.ps1]: first time I heard of it

File: src/DevOpsFlex.Automation.PowerShell/AzureKeyvaultHelpers.ps1 (hunk 2 above)
[the remark this answers is not in the document]
Comment (cjfarrelly) [reply src/DevOpsFlex.Automation.PowerShell/AzureKeyvaultHelpers.ps1]: We could put @rootik code in it's own function in here to get the short names of the regions that would ensure consistency?

File: src/DevOpsFlex.Automation.PowerShell/AzureKeyvaultHelpers.ps1 (hunk 1 above)
Comment (cjfarrelly) [reply src/DevOpsFlex.Automation.PowerShell/AzureKeyvaultHelpers.ps1]: This is something else we need a single source of truth for I think!

File: src/DevOpsFlex.Automation.PowerShell/AzureKeyvaultHelpers.ps1 (hunk 2 above)
[the remark this answers is not in the document]
Comment (VaclavK) [reply src/DevOpsFlex.Automation.PowerShell/AzureKeyvaultHelpers.ps1]: if this is the regionMap code you shared with me earlier this is different

we do not need list of ALL regions per se - we need list of evo regions (atm WE and EUS)  and for the given environment (CI being the exception) and also list of the environments too... hence that map

@djfr will book something with you @cjfarrelly  and myself to decide

[the remark this answers is not in the document]
Comment (cjfarrelly) [reply src/DevOpsFlex.Automation.PowerShell/AzureKeyvaultHelpers.ps1]: yeah, my point is that if we add more regions or env's then these types of things will have to be updated, we have them scattered across our deployment code also

[the remark this answers is not in the document]
Comment (cjfarrelly) [reply src/DevOpsFlex.Automation.PowerShell/AzureKeyvaultHelpers.ps1]: This shouldn't block a v1 though, we can get this through and backlog something to do an overall fix?

[the remark this answers is not in the document]
Comment (VaclavK) [reply src/DevOpsFlex.Automation.PowerShell/AzureKeyvaultHelpers.ps1]: I understand the concern and am happy with the conversation - let's make the call of how best to "centralise" it

**Comment by djfr** [comment]
@VaclavK let's park this for today, I will address Colin's valid concerns - got a plan for it :)
I will commit some code to the module today to deal with that in a very clean way, then you can refactor your function.

**Comment by VaclavK** [comment]
@djfr @rootik @cjfarrelly should we finalize/sign off the approach here?

**Comment by VaclavK** [comment]
@djfr 

@cjfarrelly  and @rootik , considering the PROD level KV access policy discussed yesterday, gentle ping re this

**Comment by djfr** [comment]
We have two options here:
- Just complete this PR, and tag this as technical debt
- Do the work required to start tagging the subscriptions with meta tags so we do discovery there first, and then do this: However this requires the work to tag the subscriptions to be done now and be #1 priority, we are waiting on the configuration management roll out to do the DNS test (once we can update the tool to write IP URIs, everything is just easy)

@cjfarrelly pick one and let's put this in place.

**Comment by cjfarrelly** [comment]
We have too much in the air this week to change priority, I would get this guy closed and pick it up again next week when we should be in a better place.

**Comment by djfr** [comment]
Folks, we made a decision to go down the meta RG for storing this type of information, so I'm closing down this PR.

**Pull request closed without merging**